scripts/deploy-federation.sh

```diff
@@ -67,6 +67,11 @@ function deploy-with-script() {
   # TODO(marun) Ensure federatdtypeconfig is available before creating instances
   # TODO(marun) Ensure crds are created for a given federated type before starting sync controller for that type
 
+  # Wait for the propagation of the clusterregistry CRD
+  while ! kubectl api-resources | grep clusterregistry.k8s.io > /dev/null; do
+    sleep 1
+  done
+
```

Comment:
we had an function for such kind of wait condition check `util::wait-for-condition`. Please try to use it.

Reply:
thank you. i wasn't aware of it. i'll try.

Reply:
done